### PR TITLE
Palak_Assignment23_5hrs

### PR DESCRIPTION
**Pull request by PalakSJun20**
**Program 1 :** Union Of 2 Array 
image

**Program 2 :** Intersection of 2 Array , Approach 2 ( Finding common between 2 arrays and insert unique element in another array ) 
 File Name : ArrayIntersectionMetho2.java
image

**Program 3 :** Intersection of 2 Array , Approach 1 ( Removing Duplicate element of both arrays , Comparing both arrays and assigning common value to 3rd Array ) 
File Name : ArrayIntersectionMetho1.java
Here can u pls help 

-  I created 2 methods to find unique array( array1 , array 2) if i combining getting indexoutofbound error .
- Not getting Expected Output 0.0 coming in between. 

image

### CODE REVIEW

File: src/palakSJun20/ArrayIntersectionMetho2.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+		int size = 0;
+		if (array1.length > array2.length)
+			size = array1.length;
+		else
+			size = array2.length;
+
+		double[] array3 = new double[size]; // Array 3 size is 6
+
+		// Calling Method to find Common value between 2 Array
+		findCommonArray(array1, array2, array3);
```
Comment (KrishnaKTechnocredits) [inline src/palakSJun20/ArrayIntersectionMetho2.java:28]: Why are you not using array3 which you are returning from this method?

File: src/palakSJun20/ArrayIntersectionMetho2.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+		double[] tempArray = new double[count]; // Array created with 3 size
+
+		// Removing Duplicate Element - O/P will get [10.45, 18.35, 0.0]
+		for (int outerindex = 0; outerindex < count; outerindex++) {
+			boolean flag = true;
+			for (int innerindex = 0; innerindex < outerindex; innerindex++) {
+				if (array3[outerindex] == tempArray[innerindex]) {
```
Comment (KrishnaKTechnocredits) [inline src/palakSJun20/ArrayIntersectionMetho2.java:36]: It means that here you are using wrong array3

File: src/palakSJun20/ArrayIntersectionMetho2.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+import java.util.Arrays;
+
+public class ArrayIntersectionMetho2 {
+	int count = 0;
+	int resultArrayCount = 0;
+
+	void findIntersectionOfArray(double[] array1, double[] array2) {
+		int size = 0;
+		if (array1.length > array2.length)
+			size = array1.length;
```
Comment (KrishnaKTechnocredits) [inline src/palakSJun20/ArrayIntersectionMetho2.java:21]: findIntersectionOfArray in this method here you should create min length array. You are creating max length array.

File: src/palakSJun20/ArrayIntersectionMethod1.java [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+import java.util.Arrays;
+
+public class ArrayIntersectionMethod1 {
+	int array1count = 0;
+	int array2count = 0;
+	int resultcount = 0;
+
+	void findIntersectionOfArray(double[] array1, double[] array2) {
+		double[] tempArray = new double[array1.length];
+		double[] tempArray2 = new double[array2.length];
```
Comment (KrishnaKTechnocredits) [inline src/palakSJun20/ArrayIntersectionMethod1.java:21]: Here no need to initialize this array like this.

File: src/palakSJun20/ArrayIntersectionMethod1.java [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+
+		// Calling Method to find Unique Array
+		tempArray = uniqueArray1(array1, tempArray);
```
Comment (KrishnaKTechnocredits) [inline src/palakSJun20/ArrayIntersectionMethod1.java:24]: uniqueArray1() should return tempArray.

File: src/palakSJun20/ArrayIntersectionMethod1.java [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+			if (isNumberPresent) {
+				tempArray[array1count] = array1[outerindex];
+				array1count++;
+			}
+		}
+		return tempArray; // [10.45, 14.0, 18.35, 88.88, 54.1,0.0]
+	}
+
+	// Method : Unique Array2 will give Array 2 Unique Element
+	double[] uniqueArray2(double[] array2, double[] tempArray2) {
```
Comment (KrishnaKTechnocredits) [inline src/palakSJun20/ArrayIntersectionMethod1.java:97]: Just create one method, why have you created separate methods for no. of inputs. 

double[] getUniqueInArray(double[] givenArray) {
		int count = 0;
		double[] tempArray = new double[givenArray.length];
		boolean isNumberPresent = true;

		for (int index = 0; index < givenArray.length; index++) {
			isNumberPresent = true;
			for (int innerIndex = 0; innerIndex < tempArray.length; innerIndex++) {
				if (tempArray[innerIndex] == givenArray[index])
					isNumberPresent = false;
			}
			if (isNumberPresent) {
				tempArray[count] = givenArray[index];
				count++;
			}
		}
		double[] uniqueArray = new double[count];
		for (int index = 0; index < uniqueArray.length; index++)
			uniqueArray[index] = tempArray[index];
	
		System.out.println("in sub method unique Array: " + Arrays.toString(uniqueArray));
		return uniqueArray;
	}

File: src/palakSJun20/ArrayUnion.java [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+
+public class ArrayUnion {
+	int count = 0;
+
+	void findArrayUnion(double[] array1, double[] array2) {
+		double[] tempArray3 = new double[array1.length + array2.length]; // length =12
+		
+		// Calling Method to find Unique Number Array
+		tempArray3 = findUniqueNumberArray(array1, tempArray3);
+		tempArray3 = findUniqueNumberArray(array2, tempArray3);
```
Comment (KrishnaKTechnocredits) [inline src/palakSJun20/ArrayUnion.java:20]: double[] tempUnionArray = new double[arr1.length + arr2.length];
		tempUnionArray = findUniqueNumberArray(arr2, findUniqueNumberArray(arr1, tempUnionArray));

Just write like this.

Comment (PalakSJun20) [reply src/palakSJun20/ArrayUnion.java:20]: Done

**Pull request merged**